server/aws-lsp-codewhisperer/src/language-server/netTransformServer.ts

```diff
@@ -18,7 +25,48 @@ export const NetTransformServerFactory: (
             params: ExecuteCommandParams,
             _token: CancellationToken
         ): Promise<any> => {
-            //Placeholder for logic
+            try {
+                const client = createService(credentialsProvider)
+                const transformHandler = new TransformHandler(client, workspace)
+                switch (params.command) {
+                    case 'aws/qNetTransform/startTransform': {
+                        const userInputrequest = params as QNetStartTransformRequest
+                        logging.log('prepare artifact for solution: ' + userInputrequest.SolutionRootPath)
+                        return await transformHandler.startTransformation(userInputrequest)
+                    }
+                    case 'aws/qNetTransform/getTransform': {
+                        const request = params as QNetGetTransformRequest
+                        logging.log('Calling getTransform request with job Id: ' + request.TransformationJobId)
+                        return await transformHandler.getTransformation(request)
+                    }
+                    case 'aws/qNetTransform/pollTransform': {
+                        const request = params as QNetGetTransformRequest
+                        logging.log('Calling pollTransform request with job Id: ' + request.TransformationJobId)
+                        const transformationJob = await transformHandler.pollTransformation(request)
+                        logging.log(
+                            'Transformation job for job Id' + request.TransformationJobId + ' is ' + transformationJob
+                        )
+                        return transformationJob
+                    }
+                    case 'aws/qNetTransform/getTransformPlan': {
+                        const request = params as QNetGetTransformPlanRequest
+                        logging.log('Calling getTransformPlan request with job Id: ' + request.TransformationJobId)
+                        const transformationPlan = await transformHandler.getTransformationPlan(request)
+                        logging.log(
+                            'Transformation plan for job Id' + request.TransformationJobId + ' is ' + transformationPlan
+                        )
+                        return transformationPlan
+                    }
+                    case 'aws/qNetTransform/cancelTransform': {
```

Comment:
LSP has built in cancellation, I don't think we need to add another request type for it: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#cancelRequest

Unless the cancellation might need some extra metadata to track?

Reply:
This is the cwspr cancel apis. This will be used to cancel the ongoing transformation job using the `TransformationJobId`